application/controllers/GroupController.php

```diff
@@ -225,23 +228,42 @@ public function addmemberAction()
         $this->assertPermission('config/access-control/groups');
         $groupName = $this->params->getRequired('group');
         $backend = $this->getUserGroupBackend($this->params->getRequired('backend'), 'Icinga\Data\Extensible');
-
-        $form = new AddMemberForm();
-        $form->setDataSource($this->fetchUsers())
+        $form = (new AddMemberForm())
             ->setBackend($backend)
             ->setGroupName($groupName)
             ->setRedirectUrl(
                 Url::fromPath('group/show', array('backend' => $backend->getName(), 'group' => $groupName))
             )
-            ->setUidDisabled();
+            ->setSuggestionUrl(IplUrl::fromPath(
+                'group/complete',
+                [
+                    '_disableLayout' => true,
+                    'showCompact' => true,
+                    'backend' => $this->params->getRequired('backend'),
+                    'group' => $groupName
+                ]
+            ));
 
-        try {
-            $form->handleRequest();
-        } catch (NotFoundError $_) {
-            $this->httpNotFound(sprintf($this->translate('Group "%s" not found'), $groupName));
-        }
+        $form->on(AddMemberForm::ON_SUCCESS, function ($form) {
+            $this->getResponse()->redirectAndExit($form->getRedirectUrl());
+        });
+        $form->handleRequest(ServerRequest::fromGlobals());
```

Comment:
There's `$this->getServerRequest()` now, use this instead. (Don't forget to remove the use statement)